openebs/pkg/volume/v1alpha1/volume.go

```diff
@@ -117,16 +118,16 @@ func (v CASVolume) CreateVolume(vol v1alpha1.CASVolume) error {
 	}
 	defer resp.Body.Close()
 
-	code := resp.StatusCode
-	if code != http.StatusOK {
-		return errors.New(http.StatusText(code))
-	}
-
 	data, err := ioutil.ReadAll(resp.Body)
 	if err != nil {
 		glog.Errorf("Unable to read response from maya-apiserver %v", err)
 		return err
 	}
+	code := resp.StatusCode
+	if code != http.StatusOK {
+		glog.Errorf("volume creation failed, err :%#v", string(data))
+		return fmt.Errorf(string(string(data) + ":" + http.StatusText(code)))
```

Comment:
`fmt.Errorf("%s: failed to create volume '%s': response: %+v", http.StatusText(code), vol.Name, data)`

@prateekpandey14 please check if this message works !!

Reply:
How about similar changes for delete invocation

Reply:
Above error is just a return error which has the same log msg.

Reply:
In case of delete we don't read response, similar change can't be done there.